core/login_manager.py

```diff
@@ -400,73 +405,154 @@ def is_valid(self, data):
+                distinct_id: "{user.username}"
+                client_id: BLENDER
                 }}
               )
             }}
         """
-        # print("SEND MIXPANEL:", query)
+
         try:
             request = requests.post(MixPanel.url, json={'query': query}, headers=headers)
         except Exception as e:
-            # print("ERROR:", e)
-            return
+            user.logging_in = False
+            raise Exception("No connection to the server.")
 
         if request.status_code != 200:
-            # print(f"Team API query failed to reach the server by returning code of {request.status_code}.")
-            return
+            user.logging_in = False
+            raise Exception(f"Query failed to reach the server by returning code of {request.status_code}.")
 
         # data = request.json()
-        # print("Team API response:", event_name, data)
-
-    @staticmethod
-    def send_login_event():
-        event_name = "session_start"
-        event_properties = {
-            "action": "login",
-            "blender_version": ".".join(map(str, bpy.app.version)),
-            "plugin_version": updater.current_version_str,
-        }
-        MixPanel._send_mixpanel_event(event_name, event_properties)
+        # print("MIXPANEL LOGIN RECEIVED DATA:", data)
 
     @staticmethod
     def send_logout_event():
+        headers = {"x-api-key": MixPanel.api_key}
+
         session_duration = 0
         if user.login_time:
             session_duration = datetime.datetime.utcnow().timestamp() - user.login_time
             session_duration = round(session_duration, 2)
 
-        event_name = "session_end"
         event_properties = {
             "action": "logout",
             "blender_version": ".".join(map(str, bpy.app.version)),
-            "plugin_version": updater.current_version_str,
-            "session_duration": session_duration
+            "plugin_version": user.version_str,
+            "session_duration": session_duration,
         }
-        MixPanel._send_mixpanel_event(event_name, event_properties)
+        event_properties = json.dumps(event_properties).replace("\"", "\\\"")
+
+        query = f"""
+            mutation {{
+              trackInMixpanel(input: {{
+                event_name: "session_end"
+                event_properties: "{event_properties}"
+                distinct_id: "{user.username}"
+                client_id: BLENDER
+                }}
+              )
+            }}
+        """
+
+        try:
+            request = requests.post(MixPanel.url, json={'query': query}, headers=headers)
+        except Exception as e:
+            user.logging_in = False
+            raise Exception("No connection to the server.")
+
+        if request.status_code != 200:
+            user.logging_in = False
+            raise Exception(f"Query failed to reach the server by returning code of {request.status_code}.")
+
+        # print("MIXPANEL LOGOUT QUERY:", query)
+
+    # @staticmethod
```

Comment:
The out-commented code here could also be removed (optional)

Reply:
Done!